platform/source/lib/ax/platform/platform.py

```diff
@@ -260,7 +302,10 @@ def start(self):
         # Generate kube-objects
         steps = self._config.steps
         self._load_kube_objects_from_steps(steps)
-        self._replacing = self._generate_replacing()
+        if self._cluster_config.get_cluster_provider() != ClusterProvider.USER:
+            self._replacing = self._generate_replacing()
+        else:
+            self._replacing = self._generate_default_replacing()
```

Comment:
what is the difference between `_generate_replacing ` and `_generate_default_replacing`

Reply:
Maybe ```_generate_default_replacing``` should be ```_generate_minimal_replacing``` since it is the minimal set of env variables needed to run Argo. The rest of specific to Argo provisioned clusters.

Reply:
or `_generate_replacing_for_user_provisioned_cluster`... your call how to modify the function name.